pandas_gbq/tests/test_gbq.py

```diff
@@ -968,72 +871,52 @@ def test_array_agg(self):
                 group by letter
                 order by letter ASC"""
         df = gbq.read_gbq(query, project_id=_get_project_id(),
-                          private_key=_get_private_key_path(),
+                          private_key=self.credentials,
                           dialect='standard')
         tm.assert_frame_equal(df, DataFrame([["a", [1, 3]], ["b", [2]]],
-                              columns=["letter", "numbers"]))
+                                            columns=["letter", "numbers"]))
 
 
-class TestToGBQIntegrationWithServiceAccountKeyPath(object):
+class TestToGBQIntegration(object):
     # Changes to BigQuery table schema may take up to 2 minutes as of May 2015
     # As a workaround to this issue, each test should use a unique table name.
-    # Make sure to modify the for loop range in the teardown_class when a new
+    # Make sure to modify the for loop range in the autouse fixture when a new
     # test is added See `Issue 191
     # <https://code.google.com/p/google-bigquery/issues/detail?id=191>`__
 
-    @classmethod
-    def setup_class(cls):
-        # - GLOBAL CLASS FIXTURES -
-        # put here any instruction you want to execute only *ONCE* *BEFORE*
-        # executing *ALL* tests described below.
-
-        _skip_if_no_project_id()
-        _skip_if_no_private_key_path()
-
-        _setup_common()
-
-    def setup_method(self, method):
+    @pytest.fixture(autouse=True, scope='function')
+    def setup(self, project, credentials):
```

Comment:
Ditto. I don't think we want all tests running with all credentials types, but we should have 1 test that does.